lib/widgets/draggable_scrollable_modal_bottom_sheet.dart

```diff
@@ -0,0 +1,119 @@
+import 'package:flutter/material.dart';
+
+class _DraggableScrollableLayer extends StatelessWidget {
+  const _DraggableScrollableLayer({required this.builder});
+
+  final WidgetBuilder builder;
+
+  @override
+  Widget build(BuildContext context) {
+    return DraggableScrollableSheet(
+        // Enough less than 1 as to not extend into the device's top inset
+        maxChildSize: 0.75,
+
+        // (Things break with `expand: true`, the default; not sure why.)
+        expand: false,
+
+        builder: (BuildContext context, ScrollController scrollController) {
+          return SingleChildScrollView(
+            // Prevent overscroll animation on swipe down; it looks
+            // sloppy when you're swiping to dismiss the sheet.
+            physics: const ClampingScrollPhysics(),
+
+            controller: scrollController,
+
+            child: Padding(
+              // Avoid the drag handle. See comment on
+              // _DragHandleLayer's SizedBox.height.
+              padding: const EdgeInsets.only(top: kMinInteractiveDimension),
+
+              child: Column(
+                // Extend DraggableScrollableSheet to full width so the whole
+                // sheet responds to drag/scroll uniformly. There's probably a
+                // better way to do this.
+                crossAxisAlignment: CrossAxisAlignment.stretch,
+
+                children: [
+                  Builder(builder: builder),
+                ],
+              ),
```

Comment:
Hmm yeah, I went looking for how to do this and found it somewhat tricky.

Here's a pretty clean solution, though:
```suggestion
              child: FractionallySizedBox(
                widthFactor: 1.0,
                child: Builder(builder: builder),
              ),
```

What that does is force the child to have a width that is the max width the parent allowed, while for the height passing through the same constraints (both min and max) that the parent gave.